### PR TITLE
Simplify graph loading process for services

### PR DESCRIPTION
**Pull request by kwridan**
### Short description 📝

- The graph loading process has been split into finer grained components over the past year to accomodate several ehancements to the `generate` command like, concurrent loading, mappers, plugins and more
- This has sadly made loading graphs outside of the generator a bit more verbose and error prone
    - The `tuist graph` command fails in cases where a project has a dependency in another project due to leveraging the incorrect loading sequence
- To address this, a new component (name subject to change) `SimpleGraphLoader` has been created in `TuistKit` to allow services that require the graph without any additional features to easily load by specifying a path
- The component ensures that plugins are loaded ahead of the loading process
- The `graph` acceptance tests have been enabled to help catch any future regressions to the graph command

### Test Plan 🛠

- Run `swift build && swift run tuist graph --path fixtures/ios_app_with_custom_workspace/App`
- Run `swift build && swift run tuist lint code --path fixtures/ios_app_with_custom_workspace/App`
- Run `swift build && swift run tuist lint project --path fixtures/ios_app_with_custom_workspace/App`

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://tuist.io/docs/contribution/testing-strategy/).
- [ ] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- ~In case the PR introduces changes that affect users, the documentation has been updated.~

### CODE REVIEW

**Comment by kwridan** [comment]
@fortmarek turns out the acceptance test that would have caught the `tuist graph` regression wasn't being run on CI 🤐

File: Sources/TuistGenerator/GraphViz/GraphVizGenerator.swift [hunk 1]
```diff
@@ -1,106 +0,0 @@
-import Foundation
-import GraphViz
-import TSCBasic
-import TuistCore
-import TuistGraph
-import TuistSupport
-
-public protocol GraphVizGenerating {
```
Comment (kwridan) [inline Sources/TuistGenerator/GraphViz/GraphVizGenerator.swift:8]: I've opted to removing this as after some of the fixes and simplifications, it ended up being nothing more than a proxy to `GraphToGraphVizMapping`

File: Sources/TuistKit/Utils/SimpleGraphLoader.swift [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+/// A utility for loading a graph for a given Manifest path on disk
+///
+/// - Any configured plugins are loaded
+/// - All referenced manifests are loaded
+/// - All manifests are concurrently transformed to models
+/// - A graph is loaded from the models
+///
+/// - Note: This is a simplified implementation that loads a graph without applying any mappers or running any linters
+protocol SimpleGraphLoading {
```
Comment (kwridan) [inline Sources/TuistKit/Utils/SimpleGraphLoader.swift:17]: Naming is hard 😅

Comment (pepibumur) [reply Sources/TuistKit/Utils/SimpleGraphLoader.swift:17]: Lol, I've been thinking about it for a while but I can't come
up with a good name. What about `GraphInstantiator` or `GraphWithPluginsLoader`?

**Review by pepibumur — APPROVED** [review]
This is awesome, thank you very much for tackling this @kwridan

File: Sources/TuistKit/Utils/SimpleGraphLoader.swift (hunk 2 above)
Comment (kwridan) [reply Sources/TuistKit/Utils/SimpleGraphLoader.swift:17]: maybe bring back a previously used name `ManifestGraphLoader` or `ManifestWithPluginsGraphLoader`?

**Review by fortmarek — APPROVED** [review]
Thanks a lot @kwridan! It's a great idea to extract this into its own component, possible refactoring will make it much easier plus less susceptible to regressions such as I have done 👌

File: Sources/TuistKit/Utils/SimpleGraphLoader.swift [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    /// Loads a Workspace or Project Graph at a given path based on manifest availability
+    /// - Note: This will search for a Workspace manifest first, then fallback to searching for a Project manifest
+    func loadGraph(at path: AbsolutePath) throws -> ValueGraph
+}
+
+final class SimpleGraphLoader: SimpleGraphLoading {
```
[the remark this answers is not in the document]
Comment (kwridan) [reply Sources/TuistKit/Utils/SimpleGraphLoader.swift]: 👍 

I opted for an integration test for this component as it's merely a glue of other components

**Comment by fortmarek** [comment]
tests seem to be green, can we merge this?

**Comment by pepibumur** [comment]
Ship it!

**Pull request merged**